RFCs/001-indexer.md

```diff
@@ -0,0 +1,84 @@
+The polka storage server will be extended to be part of the P2P network, connecting and sharing their peer ID on start-up. Any state changes to the Deal Database will be broadcasted to the relevant topics in the P2P network so Polka-Index can update their information.
+
+### Polka-Index
+
+Polka-Index serves as the central indexing and query service in the network, bridging users and storage providers by mapping Content Identifiers (CIDs) to the Peer IDs of nodes storing the associated data. Polka-Index specializes in organizing and disseminating metadata about where specific content can be found.
+
+Polka-Index plays a crucial role in information aggregation through the use of the [gossipsub protocol](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/README.md), a pub-sub mechanism that allows it to subscribe to specific topics. Storage providers publish deal information to these topics, enabling Polka-Index to listen for updates and build an up-to-date database of CID-to-Peer ID mappings.
+
+When an indexer node first connects to the P2P network, it catches up on existing deal information by identifying itself and requesting the current state of all deals. This is done through a dedicated gossipsub topic separate from the one used for real-time updates. By subscribing to this "catch-up" topic, the indexer can retrieve a complete snapshot of existing deal data, ensuring it starts with a full and accurate view of the network’s content before listening for new updates.
+
+By collecting and indexing this deal information, Polka-Index allows users to query the network through a JSON API, retrieving the Peer ID of a storage provider responsible for a specific CID. This approach simplifies content discovery, offering users an efficient method to locate content without requiring direct interaction with the P2P network.
+
+### Users
+
+Users in the Polka-index network act as requesters of information, leveraging the system to locate and retrieve metadata about content stored by storage providers. They query Polka-Index for mappings between Content Identifiers (CIDs) and the Peer IDs of storage providers. By doing so, users can identify the nodes responsible for storing specific data without directly interacting with the P2P network or storage providers.
+
+Through Polka-Index’s JSON API, users send simple queries containing the CID of interest and receive the associated Peer ID in response. This abstraction streamlines the content discovery process, enabling users to focus on their applications without needing to understand the complexities of the underlying network protocols.
+
+The user’s role is entirely client-facing, relying on Polka-Index as the intermediary to interact with the network, gather deal-related data, and provide accurate CID-to-Peer ID mappings. This design ensures that users experience a straightforward and efficient method for accessing decentralized storage metadata, enhancing accessibility and usability in the network.
```

Comment:
This section is a bit confusing because it obfuscates the user/client relationship when they're actually the same. For reference, the term "user" is a bit too general for this system, client/provider is much clearer